src/atomate2/openmm/jobs/base.py

```diff
@@ -0,0 +1,156 @@
+        sim, foces_indices = self._setup_base_openmm_task(input_set, output_dir)
+
+        # Run the simulation
+        sim.step(self.steps)
+
+        # Close the simulation
+        input_set = self._close_base_openmm_task()
+
+        return input_set
+
+    def _setup_base_openmm_task(self, input_set: OpenMMSet, output_dir: Optional[Union[str, Path]] = None):
+
+        # Setup compute platform and get a Simulation
+        platform_name = self.get_simulation_kwargs.get("platform")
+        platform = Platform.getPlatformByName(platform_name if platform_name else "CPU")
+        platform_props = self.get_simulation_kwargs.get("platform_properties")
+        sim = input_set.get_simulation(
+            platform=platform,
+            platformProperties=platform_props
+        )
+
+        # Add reporters
+        if self.dcd_reporter_kwargs:
+            self.dcd_reporter_kwargs["file"] = os.path.join(output_dir, self.dcd_reporter_kwargs["file"])
+            dcd_reporter = DCDReporter(**self.dcd_reporter_kwargs)
+            sim.reporters.append(dcd_reporter)
+        if self.state_reporter_kwargs:
+            self.state_reporter_kwargs["file"] = os.path.join(output_dir, self.state_reporter_kwargs["file"])
+            state_reporter = StateDataReporter(**self.state_reporter_kwargs)
+            sim.reporters.append(state_reporter)
+        if self.pdb_reporter_kwargs:
+            self.pdb_reporter_kwargs["file"] = os.path.join(output_dir, self.pdb_reporter_kwargs["file"])
+            pdb_reporter = DCDReporter(**self.pdb_reporter_kwargs)
+            sim.reporters.append(pdb_reporter)
+
+        # Add barotats
+        force_indices = list()
+        context = sim.context
+        system = context.getSystem()
+        if self.temperature_barostat_kwargs:
+            thermostat_force_index = system.addForce(
+                AndersenThermostat(self.temperature * kelvin, pico * second * self.frequency)
+            )
+            context.reinitialize(preserveState=True)
+            force_indices.append(thermostat_force_index)
+        if self.pressure_barostat_kwargs:
+            assert (
+                system.usesPeriodicBoundaryConditions()
+            ), "system must use periodic boundary conditions for pressure equilibration."
+            barostat_force_index = system.addForce(
+                MonteCarloBarostat(self.pressure * atmosphere, self.temperature * kelvin, 10)
+            )
+            context.reinitialize(preserveState=True)
+            force_indices.append(barostat_force_index)
+
+        return sim, force_indices
+
+    def _close_base_openmm_task(self, context: Context, force_indices: Optional[list]):
+        # output_set = copy.deepcopy(input_set)
+        state = StateInput(
+            context.getState(
+                getPositions=True,
+                getVelocities=True,
+                enforcePeriodicBox=True,
+            )
+        )
+
+        input_set[input_set.state_file] = state
```

Comment:
Instead of including a PDB reporter, let's update the pdb file in the output input_set. We can do that by attaching a reporter, asking it to instantly report, and then serializing that as a TopologyInput. This let's us get an updated pdb file (which has utility) without writing out a whole bunch during the simulation.

Happy to help if you need it.